include/boost/gil/point.hpp

```diff
@@ -0,0 +1,236 @@
+//
+// Copyright 2005-2007 Adobe Systems Incorporated
+//
+// Distributed under the Boost Software License, Version 1.0
+// See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt
+//
+#ifndef BOOST_GIL_POINT_HPP
+#define BOOST_GIL_POINT_HPP
+
+#include <boost/gil/utilities.hpp>
+
+#include <boost/config.hpp>
+#include <boost/config/no_tr1/cmath.hpp>
+
+#include <cstddef>
+
+namespace boost { namespace gil {
+
+/// \addtogroup PointModel
+///
+/// Example:
+/// \code
+/// point2<std::ptrdiff_t> p(3,2);
+/// assert((p[0] == p.x) && (p[1] == p.y));
+/// assert(axis_value<0>(p) == 3);
+/// assert(axis_value<1>(p) == 2);
+/// \endcode
+
+/// \brief 2D point both axes of which have the same dimension type
+/// \ingroup PointModel
+/// Models: Point2DConcept
+template <typename T>
+class point2
+{
+public:
+    using value_type = T;
+
+    template<std::size_t D>
+    struct axis
+    {
+        using coord_t = value_type;
+    };
+
+    static constexpr std::size_t num_dimensions = 2;
+
+    point2() = default;
+    point2(T px, T py) : x(px), y(py) {}
+    point2(point2 const& p) : x(p.x), y(p.y) {}
```

Comment:
This is not required and makes the type non-trivial even when its members are.
```suggestion
```

Reply:
Remove superfluous copy constructor from trivial point2 class